audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -139,221 +146,238 @@
+        var requests = GOOD_SUCCESS_RECORD_NODE_FACTORY.get().limit(3L);
 
         // When - we write to the input
-        ProducerSettings<String, JsonNode> producerSettings = ProducerSettings
-                .create(akkaActorSystem, new StringSerializer(), new RequestSerializer())
-                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
-
-        Source.fromJavaStream(() -> requests)
-                .runWith(Producer.plainSink(producerSettings), akkaMaterializer)
-                .toCompletableFuture().join();
-
+        runStreamOf(requests);
         waitForService();
 
         // Then - check the audit service has invoked the audit method 3 times
-        Mockito.verify(auditService, Mockito.timeout(3000).times(3)).audit(anyString(), any());
+        verify(auditService, timeout(3000).times(3)).audit(anyString(), any());
     }
 
     @Test
     @DirtiesContext
     void testGoodAndBadSuccessRequestSet() throws InterruptedException {
+
         // Add 2 `Good` and 2 `Bad` success messages to the success topic
         // The ContractTestData.REQUEST_TOKEN maps to partition 0 of [0, 1, 2], so the akka-test yaml connects the consumer to only partition 0
-        final Stream<ProducerRecord<String, JsonNode>> requests = Stream.of(
-                ContractTestData.GOOD_SUCCESS_RECORD_NODE_FACTORY.get().limit(1L),
-                ContractTestData.BAD_SUCCESS_RECORD_NODE_FACTORY.get().limit(2L),
-                ContractTestData.GOOD_SUCCESS_RECORD_NODE_FACTORY.get().limit(1L))
-                .flatMap(Function.identity());
+        var requests = Stream.of(
+                GOOD_SUCCESS_RECORD_NODE_FACTORY.get().limit(1L),
+                BAD_SUCCESS_RECORD_NODE_FACTORY.get().limit(2L),
+                GOOD_SUCCESS_RECORD_NODE_FACTORY.get().limit(1L))
+            .flatMap(Function.identity());
 
         // When - we write to the input
-        ProducerSettings<String, JsonNode> producerSettings = ProducerSettings
-                .create(akkaActorSystem, new StringSerializer(), new RequestSerializer())
-                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
-
-        Source.fromJavaStream(() -> requests)
-                .runWith(Producer.plainSink(producerSettings), akkaMaterializer)
-                .toCompletableFuture().join();
-
+        runStreamOf(requests);
         waitForService();
 
         // Then - check the audit service has invoked the audit method for the 2 `Good` requests
-        Mockito.verify(auditService, Mockito.timeout(3000).times(2)).audit(anyString(), any());
+        verify(auditService, timeout(3000).times(2)).audit(anyString(), any());
     }
 
     @Test
     @DirtiesContext
-    void testRestEndpointForErrorMessage() throws InterruptedException {
+    void testRestEndpointForErrorMessage(
+            @Autowired final TestRestTemplate restTemplate) throws InterruptedException {
```

Comment:
I understood that with a few exceptions, MockMvc was the first choice over TestRestTemplate. Why are we using this here?

Reply:
I have no idea. i did not write that, but I will investigate.

Reply:
So it's confirmed that for contract tests, `TestRestTemplate` should be used with the environment uisng a random port. This will then get Spring to start a real HTTP server and not a Mock one.